Nasal/engine.nas

```diff
@@ -87,39 +87,73 @@ var primerTimer = maketimer(5, func {
 });
 
 # ========== oil consumption ======================
+# Thanks to HHS81 for more advanced simulation
+var service_hours = getprop("/engines/active-engine/oil-service-hours");
+var consumption_qph = 0.0;
+var rpm_factor = 0.0;
+var rpm = 0;
+var oil_level = 0;
+var oil_full = 0;
+var oil_lacking = 0;
+var oil_level_limited = 0;
+var service_hours_increase = 0;
+var service_hours_new = 0;
+var low_oil_pressure_factor = 0.0;
+var low_oil_temperature_factor = 0.0;
 
 var oil_consumption = maketimer(1.0, func {
-
-    var oil_level = getprop("/engines/active-engine/oil-level");
+	
+	oil_level = getprop("/engines/active-engine/oil-level");
     if (getprop("/controls/engines/active-engine") == 0)
-        var oil_full = 7;
+        oil_full = 7;
     if (getprop("/controls/engines/active-engine") == 1)
-        var oil_full = 8;
-    var oil_lacking = oil_full - oil_level;
+        oil_full = 8;
+    oil_lacking = oil_full - oil_level;
     setprop("/engines/active-engine/oil-lacking", oil_lacking);
     
     if (getprop("/engines/active-engine/oil_consumption_allowed")) {
     
-        var rpm = getprop("/engines/active-engine/rpm");
+        rpm = getprop("/engines/active-engine/rpm");
     
         # Quadratic formula which outputs 1.0 for input 2300 RPM (cruise value),
         # 0.6 for 700 RPM (idle) and 1.2 for 2700 RPM (max)
-        var rpm_factor = 0.00000012 * math.pow(rpm, 2) - 0.0001 * rpm + 0.62;
-    
-        # Consumption rate defined as 1.5 quarter per 10 hours (36000 seconds)
-        # at cruise RPM
-        var consumption_rate = 1.5 / 36000; 
+        rpm_factor = 0.00000012 * math.pow(rpm, 2) - 0.0001 * rpm + 0.62;
     
-        if (getprop("/engines/active-engine/running")) {
-            oil_level = oil_level - consumption_rate * rpm_factor;
-            setprop("/engines/active-engine/oil-level", oil_level);
+        # Consumption rate defined as 0.33 quarts per 1 hour (3600 seconds) (Lycoming Manual 3-6 p27)
+        # at 2350 RPM (normal cruise)
+        consumption_qph = 0.33 / 3600; 
+		
+		# Raise consumption when oil level is > 8 quarts (blowout)
+        if (oil_level > oil_full) {
+            consumption_qph = consumption_qph * 1.3;
         }
+        
+        # Consumption also raises with oil in service time (lower viscosity => more friction)
+        # (Oil should be changed at 50 hrs!)
+        # See: http://www.t-craft.org/Reference/Aircraft.Oil.Usage.pdf
+        # Hours:        0 |    10 |    25 |  50   |    75
+        # Add Qts/hr:   0 |  0.02 | 0.125 | 0.5   | 1.125
+		service_hours = getprop("/engines/active-engine/oil-service-hours");
+        service_hours_increase = 0.00020 * math.pow(service_hours, 2);
+        if (service_hours_increase > 1.5) service_hours_increase = 1.5;  # cap at that rate
```

Comment:
Can you use { } here? You can also write `service_hours_increase = std.min(1.5, service_hours_increase);`